Fallback resolver

niner opened the pull request:
When we run code at BEGIN time the actual lexical environment the code
is supposed to run in hasn't been set up yet. Previously we got around
that by compiling a wrapper frame, containing all lexicals visible from
the code we run. This frame is rather expensive and we want to replace
it with an on-demand resolver that's part of the compiler (i.e. HLL
code)

niner committed:
Fallback resolvers for compunits and syscall for setting them

When we run code at BEGIN time the actual lexical environment the code
is supposed to run in hasn't been set up yet. Previously we got around
that by compiling a wrapper frame, containing all lexicals visible from
the code we run. This frame is rather expensive and we want to replace
it with an on-demand resolver that's part of the compiler (i.e. HLL
code)

niner committed:
Support a fallback resolver when looking for dynamic variables

niner committed:
Support fallback resolver in getlexouter

niner committed:
Support fallback resolver in *getlex*

jnthn reviewed (commented):
Makes sense to me. Is there any change needed in `oplist`, so indicate that the various lexical lookup ops might now end up invoking?

jnthn commented on src/core/oplist:
```diff
@@ -89,17 +89,17 @@ DEPRECATED_63       r(obj) ins
-getlex_nn           w(num64) str :pure :noinline
-getlex_ns           w(str) str :pure :noinline
-getlex_no           w(obj) str :pure :deoptonepoint :logged :noinline
+getlex_ni           w(int64) str :pure :noinline :invokish :throwish
+getlex_nn           w(num64) str :pure :noinline :invokish :throwish
+getlex_ns           w(str) str :pure :noinline :invokish :throwish
+getlex_no           w(obj) str :pure :deoptonepoint :logged :noinline :invokish :throwish
```
If they can call now, I wonder if also `deoptallpoint` applies?

niner commented:
> If they can call now, I wonder if also `deoptallpoint` applies?

Wouldn't the same be true then for assign and decont which can also call?

jnthn commented:
> Wouldn't the same be true then for assign and decont which can also call?

Yes. Seems `istype` carires it, at least.

niner committed:
Add missing :deoptallpoint flags to invokish ops

Since invoked code may do anything, including creating mixins, we have
to mark all non-spesh ops that may invoke as deoptallpoints.
